### ISSUE TITLE
Meta data 

### ISSUE THREAD

**Issue by faizan1234567** [pyg-team/pytorch_geometric#4452]
### 🐛 Describe the bug

I defined HAN class for heterogeneous graph learning on the IMDB dataset. However, when I create metadata and pass it on to the HAN class it gives me " __init__() got multiple values for argument 'metadata' ". The code is shown here.

path = osp.join(osp.dirname(osp.realpath('__file__')), '../../data/IMDB')

metapaths = [[('movie', 'actor'), ('actor', 'movie')],
             [('movie', 'director'), ('director', 'movie')]]
transform = T.AddMetaPaths(metapaths= metapaths, drop_orig_edges= True, drop_unconnected_nodes= True)
dataset = IMDB(path, transform= transform)
data = dataset[0]

class HAN(nn.Module):
  def __init__(self, in_channels: Union[int, Dict[str, int]],
               out_channels: int, hidden_channels = 128, heads=8):
    super().__init__()

    self.HanConv = HANConv(in_channels, hidden_channels, heads, 
                           dropout = 0.6, metadata=data.metadata())
    self.lin = nn.Linear(hidden_channels, out_channels)

  def forward(self, x_dict, edge_index_dict):
    x = self.HanConv(x_dict, edge_index_dict)
    x = self.lin(x['movie'])
    return x
model = HAN(in_channels= -1, out_channels= 3)
print(model)

please help me fix this bug.

### Environment

* PyG version:
* PyTorch version:
* OS:
* Python version:
* CUDA/cuDNN version:
* How you installed PyTorch and PyG (`conda`, `pip`, source):
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
Can you define via `HANConv(in_channels, hidden_channels, heads=heads, …)`? This should fix the issue.

**Comment by faizan1234567** [comment]
Sure let me check

**Comment by faizan1234567** [comment]
Now it gives me this error.
![han_Conv](https://user-images.githubusercontent.com/61932757/162699394-42a6eedd-95fd-40a7-a65e-8cd979d41110.png)

**Comment by faizan1234567** [comment]
I am using [this](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/han_imdb.py) example from PyTorch geometric examples.

**Comment by faizan1234567** [comment]
Moreover, I am running my code on Google Colab.

**Comment by faizan1234567** [comment]
please see this as well.
![Uploading han_conv1.png…]()

**Comment by faizan1234567** [comment]
please see this as well.
![han_conv1](https://user-images.githubusercontent.com/61932757/162700420-ffe6e424-4a26-49ea-bff6-6aa785abc0d5.png)

**Comment by faizan1234567** [comment]
Ok issue resolved. I used HANConv(in_channels, hidden_channels, heads = heads, dropout = 0.6, metadat = data.metadata())...
this solved the issue. 
Thank you for your support
